providers/implementations/ciphers/cipher_enull_hmac.c

```diff
@@ -0,0 +1,343 @@
+static OSSL_FUNC_cipher_get_params_fn enull_hmac_##lcmd##_get_params;          \
+static int enull_hmac_##lcmd##_get_params(OSSL_PARAM params[])                 \
+{                                                                              \
+    return ossl_cipher_generic_get_params(params, 0, flags,                    \
+                                          kbits, blkbits, ivbits);             \
+}                                                                              \
+                                                                               \
+static OSSL_FUNC_cipher_newctx_fn enull_hmac_##lcmd##_newctx;                  \
+static void *enull_hmac_##lcmd##_newctx(void *provctx)                         \
+{                                                                              \
+    PROV_ENULL_HMAC_CTX *ctx;                                                  \
+                                                                               \
+    if (!ossl_prov_is_running())                                               \
+        return NULL;                                                           \
+                                                                               \
+    if ((ctx = OPENSSL_zalloc(sizeof(*ctx))) == NULL)                          \
+        return NULL;                                                           \
+                                                                               \
+    if ((ctx->hmac = HMAC_CTX_new()) == NULL) {                                \
+        OPENSSL_free(ctx);                                                     \
+        return NULL;                                                           \
+    }                                                                          \
+    if (ossl_prov_digest_fetch(&ctx->md, PROV_LIBCTX_OF(provctx),              \
+                               #UCMD, NULL) == NULL) {                         \
+        enull_hmac_freectx(ctx);                                               \
+        return NULL;                                                           \
+    }                                                                          \
+    ctx->tag_len = ENULL_HMAC_##UCMD##_TAGLEN;                                 \
+    ossl_cipher_generic_initkey(ctx, kbits, blkbits, ivbits, 0, flags,         \
+                                ossl_prov_cipher_hw_enull_hmac(kbits),         \
+                                provctx);                                      \
+    return ctx;                                                                \
+}                                                                              \
+                                                                               \
+const OSSL_DISPATCH ossl_enull_hmac_##lcmd##_functions[] = {                   \
+    { OSSL_FUNC_CIPHER_NEWCTX, (void (*)(void))enull_hmac_##lcmd##_newctx },   \
```

Comment:
Hint: instead of repeating those ugly`(void (*)(void)` casts, e.g. define
```
#define OSSL_DISPATCHALG(num, name) {OSSL_FUNC_CIPHER_##num, (void (*)(void)) name}
```
and use, e.g.,
```
OSSL_DISPATCHALG(NEWCTX, enull_hmac_##lcmd##_newctx);
```